src/server/ua_subscription.h

```diff
@@ -81,7 +81,8 @@ struct UA_MonitoredItem {
     UA_UInt32 maxQueueSize;
     UA_Boolean discardOldest;
     // TODO: dataEncoding is hardcoded to UA binary
-    UA_DataChangeTrigger trigger;
+    UA_DataChangeFilter filter;
+    UA_Variant lastValue;
```

Comment:
Can't we reuse `lastSampledValue`?

Reply:
This is not possible. lastSampleValue is a Byterepresentation of the value and will be string compared. If it is not equal, it will be reported to the client. For me this is basiclly a hack which helps to compare values type independant. Here for filters you have to check which type is in the variant and handle it properly. This could be done for other not numeric types, too, but it isn't done yet. If someone implement it, it is possible to remove lastSampledValue and use the variant. Until then I would let it like it is in this PR.

Reply:
Let's explore the performance impact.
If it is acceptable I would also prefer to have a "single source of truth" for the last value.
In a next PR if we decide to do this.